samples/create_vm.py

```diff
@@ -0,0 +1,210 @@
+
+    parser.add_argument('-v', '--vmdk-file',
+                        required=True,
+                        action='store',
+                        help='Path on datastore to .vmdk file')
+
+    parser.add_argument('-n', '--name',
+                        required=True,
+                        action='store',
+                        help='Name of VM to create')
+
+    parser.add_argument('-m', '--memory',
+                        type=int,
+                        required=True,
+                        action='store',
+                        help='Memory in MB')
+
+    parser.add_argument('-g', '--guest_type',
+                        required=True,
+                        action='store',
+                        help='VM guest type')
+
+    args = parser.parse_args()
+
+    return cli.prompt_for_password(args)
+
+
+def find_vm(service_instance, name):
+    obj = None
+
+    content = service_instance.RetrieveContent()
+    container = content.viewManager.CreateContainerView(
+        content.rootFolder, [vim.VirtualMachine], True)
+    for c in container.view:
+        if c.name == name:
+            obj = c
+            break
+    return obj
+
+
+def create_vm(vm_name, vmdk_path, vm_ram, vm_guest,
+              service_instance, data_store,
+              vm_folder, resource_pool):
+    """Creates a VirtualMachine.
+
+    :param vm_name: String Name for the VirtualMachine
+    :param vmdk_path: Full path to vmdk file in the datastore
+    :param vm_ram: Memory in MB
+    :param vm_guest: Guest type, e.g. rhel7_64Guest
+    :param service_instance: ServiceInstance connection
+    :param data_store: DataStore to place the VirtualMachine on
+    :param vm_folder: Folder to place the VirtualMachine in
+    :param resource_pool: ResourcePool to place the VirtualMachine in
+    """
+    # bare minimum VM shell, no disks.
+    vmx_path = '[' + data_store.info.name + '] ' + vm_name
+    vmx_file = vim.vm.FileInfo(logDirectory=None,
+                               snapshotDirectory=None,
+                               suspendDirectory=None,
+                               vmPathName=vmx_path)
+
+    # config for VM
+    config = vim.vm.ConfigSpec(name=vm_name, memoryMB=vm_ram, numCPUs=1,
+                               files=vmx_file, guestId=vm_guest,
+                               version='vmx-13')
+
+    print("Creating VM {}...".format(vm_name))
+    task = vm_folder.CreateVM_Task(config=config, pool=resource_pool)
+    tasks.wait_for_tasks(service_instance, [task])
+
+    vm = find_vm(service_instance, vm_name)
+
+    # find IDE 0 controller to attach disk to
+    controller = None
+    for dev in vm.config.hardware.device:
+        if isinstance(dev, vim.vm.device.VirtualIDEController) and \
+           dev.deviceInfo.label == 'IDE 0':
+            controller = dev
+
+    disk_spec = vim.vm.device.VirtualDeviceSpec(
+        device=vim.vm.device.VirtualDisk(
+            unitNumber=1,
```

Comment:
I tried the script and the error I received (I had to look at the vmware.log file of the VM) was:
2018-10-31T17:56:52.615Z| vmx| I125+ The IDE device (disks/CD-ROM) configuration is incorrect. There is an IDE slave with no master at 'ide0:1'. This configuration does not work correctly in virtual machines. Move the disk/CD-ROM from 'ide0:1' to 'ide0:0' using the configuration editor.

If you set this to unitNumber=0, then I could power on.
(Alternatively you could attach another IDE device in slot 0 as the master).

Reply:
Thank you, power on works now.